### DISCUSSION TITLE
Roadmap

### DISCUSSION THREAD

**Discussion by aolko** [tauri-apps/tauri#1151] [Q&A] [answered]
It's already Q1 2021 and some people might have the following questions:
- Where's the beta release? (Q4 2020)
- Where's (or, rather, when exactly) the stable release? (Q1 2021)
- Why don't you watch your own roadmap?

**Edit:** Wow, this discussion got a burst of sensitive responses. Well, i guess i won't touch the topic of taking **full responsibility** for your project then, not to mention that curiosity killed the cat.

**Comment by MrOggy85** [comment]
Same question from me! I really like Tauri and find it super useful. But, I would like to enhance the app by adding menus etc to make it feel like a proper native desktop app. How can I do that?

**Reply by nothingismagick** [reply]
This is on our to do list. If you are able to support the work on this, then please join the Discord server.

**Comment by nothingismagick** [comment, marked as answer]
You know, I sort of avoided this question because I felt like it was really counter-productive and virtually passive-aggressive. Instead of trash-talking, though, I will use the opportunity to explain to you, @aolko , where I think your tone of voice missed the mark and how I think you could have gotten a more immediate and friendly response:

First of all, it sounds like you are using the classically "western white-male privilege" tone of voice, although your grammar is slightly off - which tells me you are not a native speaker. Following some of your other contributions this year at some PHP projects confirms that you do indeed have a ways to go in your mastery of the English language.

Now, I am not exactly sure where we hurt your feelings, because it seems to me like you think we lied to you and broke some kind of contract. I am REALLY sorry if you thought that a roadmap is something written in stone. (It is actually [written in JSON](https://github.com/tauri-apps/tauri-docs/blob/8d98bbaf5c1291071ba7c2f0224e153edb4a4f9c/translations/en.json#L43).) I would be lying if I told you that we didn't use it as a way to get people off our back who asked us every 10 days if feature Y was going to get merged.

And finally, we can get to the core issue, and we have all been there. Feeling sad that some open source project that you depend on / want to depend on hasn't released a feature you need within the timeframe that you expected it. I get it. It can be really, really frustrating. I am not going to give you the classic spiel of "just open a PR", because were you capable of doing that you would have already done such a thing (I am sure of it 😄  ) I don't want to scare you away from potentially becoming a contributor in the future.

But just for a second, imagine you wanted to play çyberpünk 2O77: you waited years and years for it - and then when it arrived, you felt totally betrayed when collision detection just didn't work as well as your beloved RDR2. Now, to bring this back to Tauri - this is software engineering, which has complications anyway. But the entire team is made of unpaid volunteers, not some mythical highly-paid 1000-person crew of CD Projekt. Delays happen. Deal with it. We want something great and won't release air-swimming, leg-twitching or trunk-surfing and call it a day.

So next time you are feeling sad about Tauri, please try to remember that there are PEOPLE, like real HUMAN BEINGS, on the other side of your screen who do actually care very much about this project and are doing their best in a very hostile world. Now, in this context, please ask yourself - does your `Why don't you watch your own roadmap?` question still make sense? If it doesn't, I invite you to edit your question in a more friendly and accepting way.

**Comment by nothingismagick** [comment]
Full disclosure: We banned @aolko from our Discord almost exactly one year ago for similarly disparaging remarks that violated our community standards. Passive aggressiveness is just totally out of line. This is not Twitter.
